src/api.ts

```diff
@@ -35,127 +38,54 @@ export class ScenarioApi {
-        const configSeed: T.ConfigSeed = configs[playerName]
-        if (!_.isFunction(configSeed)) {
-          throw new Error(`Config for player '${playerName}' contains something other than a function. Either use Config.gen to create a seed function, or supply one manually.`)
-        }
-
-        const partialConfigSeedArgs = trycp
-          ? await trycp.setup(playerName)
-          : await localConfigSeedArgs()
-        const configSeedArgs: T.ConfigSeedArgs = _.assign(partialConfigSeedArgs, {
-          scenarioName: this.description,
-          playerName,
-          uuid: this._uuid
-        })
-        logger.debug('api.players: seed args generated for %s = %j', playerName, configSeedArgs)
-        const configJson = await configSeed(configSeedArgs)
-        logger.debug("built config: %s", JSON.stringify(configJson))
-        configsJson.push(configJson)
-
-        if (!configJson.environment_path) {
-          throw new Error("Generated config does not have environment_path set")
-        }
-
-        // this code will only be executed once it is determined that all configs are valid
-        playerBuilders[playerName] = async () => {
-          const configDir = configJson.environment_path
-
-          // FIXME: can we get this from somewhere?
-          const adminInterfacePort = 0
-
-          if (trycp) {
-            const newConfigJson = await interpolateConfigDnaUrls(trycp, configJson)
-            await trycp.player(playerName, newConfigJson)
-          } else {
-            await fs.writeFile(getConfigPath(configDir), YAML.stringify(configJson))
-          }
-          logger.debug('api.players: player config committed for %s', playerName)
-
-
-          const player = new Player({
-            name: playerName,
-            config: configJson,
-            configDir,
-            adminInterfacePort,
-            spawnConductor,
-            onJoin: () => console.log("FIXME: ignoring onJoin"),//instances.forEach(instance => this._waiter.addNode(instance.dna, playerName)),
-            onLeave: () => console.log("FIXME: ignoring onLeave"),//instances.forEach(instance => this._waiter.removeNode(instance.dna, playerName)),
-            onActivity: () => this._restartTimer(),
-            onSignal: (signal_data) => {
-              console.log("ignoring signal:", signal_data)
-/*              const instance = instances.find(c => c.id === instanced)
-              const dnaId = instance!.dna
-              const observation = {
-                dna: dnaId,
-                node: playerName,
-                signal
-              }
-              this._waiter.handleObservation(observation)
-*/
-            },
-          })
-          return player
-        }
+    })
+
+    // create all the promise *creators* which will
+    // create the players
+    const playerBuilders: Array<PlayerBuilder> = await Promise.all(playerConfigs.map(
+      async configSeed => {
+        const playerName = uuidGen()
```

Comment:
This has a significant downside, that the debug logs have a crazy UUID per conductor rather than the name of the player.